packages/framework/src/Component/Enum/AbstractEnum.php

```diff
@@ -13,7 +13,15 @@ class AbstractEnum
      */
     public function getAllCases(): array
     {
-        return ReflectionHelper::getAllPublicClassConstants(static::class);
+        $publicConstants = ReflectionHelper::getAllPublicClassConstants(static::class);
+
+        foreach ($publicConstants as $case) {
+            if (in_array($case, $this->getUnusedConstants(), true)) {
+                unset($publicConstants[array_search($case, $publicConstants, true)]);
+            }
+        }
+
+        return $publicConstants;
```

Comment:
```suggestion
return array_diff($publicConstants, $this->getUnusedConstants()); 
```

This should do the trick as well, looks more simple, what do you think?